File: content/snippets/cpp/language_basics_file_io.cpp
#include <fstream>
#include <iostream>
#include <string>
#include <vector>

int main() {
    const std::string path = "file_io_demo.txt";

    {
        std::ofstream out(path);
        out << "apple\nbanana\ncarrot\n";
    }

    std::ifstream in(path);
    std::vector<std::string> lines;
    for (std::string line; std::getline(in, line);) {
        lines.push_back(line);
    }

    std::cout << "read lines: ";
    for (const auto& line : lines) {
        std::cout << line << ' ';
    }
    std::cout << "\n";
    return 0;
}

File: content/snippets/typescript/language_basics_file_io.ts
#!/usr/bin/env ts-node

import { setImmediate } from "node:timers";
import { promises as fs } from "fs";

async function main(): Promise<void> {
  const path = "file_io_demo.txt";
  await fs.writeFile(path, "apple\nbanana\ncarrot\n", "utf8");

  const content = await fs.readFile(path, "utf8");
  const lines = content.trim().split(/\r?\n/);
  console.log("read lines:", lines);
}

main()
  .then(() => setImmediate(() => {}))
  .catch((err) => {
    console.error("error:", err);
    process.exit(1);
  });
